core/utils/format.cc

```diff
@@ -0,0 +1,34 @@
+#include "format.h"
+
+#include <cassert>
+#include <memory>
+
+namespace bess {
+namespace utils {
+
+std::string Format(const char *fmt, ...) {
+  va_list ap;
+  va_start(ap, fmt);
+  const std::string s = FormatVarg(fmt, ap);
+  va_end(ap);
+  return s;
+}
+
+std::string FormatVarg(const char *fmt, va_list ap) {
+  const int init_bufsize = 128;
+  std::unique_ptr<char[]> buf;
+
+  buf.reset(new char[init_bufsize]);
+  int ret = vsnprintf(&buf[0], init_bufsize, fmt, ap);
```

Comment:
(It's true that vasprintf() is a GNU/Linux/BSD function, not standard C, so that could be a reason not to use it.)